crypto/curve25519.go

```diff
@@ -123,6 +123,17 @@ func ed25519Verify(public ed25519PublicKey, data []byte, sig ed25519Signature) b
 	return result == 0
 }
 
+func ed25519VerifyBatchVerificationCompatibleVersion(public ed25519PublicKey, data []byte, sig ed25519Signature) bool {
+	// &data[0] will make Go panic if msg is zero length
+	d := (*C.uchar)(C.NULL)
+	if len(data) != 0 {
+		d = (*C.uchar)(&data[0])
+	}
+
+	result := C.crypto_sign_ed25519_bv_compatible_verify_detached((*C.uchar)(&sig[0]), d, C.ulonglong(len(data)), (*C.uchar)(&public[0]))
```

Comment:
could you have some type of GC testing to make sure that the GC won't collect `d` prematurely ?